### PR TITLE
Adding pip-6, In-Memory Storage of Public Keys, Validators, and Accounts using LRU Cache

### PR DESCRIPTION
**Pull request by Ja7ad**

### CODE REVIEW

File: PIPs/pip-6.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+3. **Account Information**: Account data will be cached when accounts are accessed or updated. This will reduce the need for repeated disk reads when handling transactions involving specific accounts.
+
+
+## Security Considerations
+
+To ensure the security of the LRU cache, we need to consider the following:
+
+1. **Cache Invalidation**: Proper mechanisms must be in place to invalidate cache entries when data is updated or removed. This ensures that outdated or incorrect data is not retrieved from the cache.
+
+2. **Concurrency**: Handling concurrent access to the cache is crucial. We should implement appropriate locking mechanisms to prevent race conditions and data corruption.
```
Comment (b00f) [inline PIPs/pip-6.md:44]: Indeed the store has its locking mechanism.

**Comment by b00f** [comment]
Good job @Ja7ad

File: PIPs/pip-6.md (hunk 1 above)
Comment (Ja7ad) [reply PIPs/pip-6.md:44]: Yes, If we need to implement LRU Cache without a module, required use is the lock mechanism.

**Review by b00f — APPROVED** [review]

File: PIPs/pip-6.md (hunk 1 above)
Comment (parsa-hedayatnia) [reply PIPs/pip-6.md:44]: Implementing an LRU cache for in-memory storage of public keys, validators, and account data has the potential to offer performance benefits and reduce storage requirements, it also introduces complexities and potential risks that need to be carefully managed. Proper cache management, cache invalidation, and concurrency control are crucial aspects of this proposal.

**Review by parsa-hedayatnia — APPROVED** [review]

**Pull request merged**